graph/graph.go

```diff
@@ -186,27 +171,137 @@ check:
 		}
 		last = bar
 	}
+
+	return matrix
+}
+
+// GenerateGraph with ASCII graph with ANSI escapes
+func GenerateGraph(chart *api.Chart, width int, height int, chartTheme ChartTheme, timezone *time.Location) (string, error) {
+	ran := chart.High.Sub(chart.Low)
+
+	maxSize := len(strings.Split(chart.High.String(), ".")[0]) + 3
+
+	spacing := (width) / (chart.Length)
+
+	if spacing == 0 {
+		spacing = 3
+	}
+
+	out := "┏"
+
+	borderHorizontal(&out, width+maxSize+3)
+	out += "┓"
+	colour := 92
+	if chart.Length < width/5 {
+		chart.Length = width / 3
+	}
+	if chart.Length > width {
+		mod := 2 * (chart.Length / width)
+		chartTemp := make([]*api.Bar, 0)
+		for i, bar := range chart.Bars {
+			if (i+1)%mod == 0 {
+				chartTemp = append(chartTemp, bar)
+			}
+		}
+		chart.Bars = chartTemp
+		chart.Length = len(chartTemp)
+	}
+	if chart.Change.IsNegative() {
+		colour = 91
+	}
+	info := fmt.Sprintf(
+		"\n┃\033[95m %s | \033[%dm%s %s (%s%% | %s)\033[95m on %s | ",
+		chart.Ticker,
+		colour,
+		chart.Close.StringFixed(2),
+		chart.Currency,
+		chart.Change.StringFixed(2),
+		chart.ChangeVal.StringFixed(2),
+		chart.End.Time().Format(dateFormat),
+	)
+	if len(info) > width {
+		info += fmt.Sprintf(
+			"%s \033[0m",
+			chart.Exchange,
+		)
+	} else {
+		info += fmt.Sprintf(
+			"Prev: %s | %s \033[0m",
+			chart.Prev.StringFixed(2),
+			chart.Exchange,
+		)
+	}
+
+check:
+	if len(info) < width+maxSize+24 {
+		info += " "
+		goto check
+	}
+	info += "┃\n┣"
+	out += info
+	borderHorizontal(&out, width+maxSize+3)
+
+	out += "┫"
+
 	increment := ran.Div(decimal.NewFromInt(int64(height)))
-	for i, slc := range matrix {
-		out += "┃"
-		price := chart.High.Sub(increment.Mul(decimal.NewFromInt(int64(i)))).StringFixed(2)
-	checkLen:
-		if len(price) < maxSize {
-			price = " " + price
-			goto checkLen
+	out += "\n"
+
+	if chartTheme == Braille {
```

Comment:
Using the package also introduces extra logic here some of which isn't 100% DRY. Could you implement the packages functionality in a more cohesive way inside the package? Cheers :)